tilauskasittely/tilaus_myynti.php

```diff
@@ -4739,6 +4739,40 @@ function update_summa(kaikkiyhteensa) {
           $_ei_jt_meilia = 'X';
         }
         elseif ($tapa != "POISJTSTA" and $tapa != "PUUTE" and $tapa != "JT") {
+          if ($tapa == "POISTA") {
+            // Mik�li tilausriviin liittyy ostorivi, niin poistetaan my�s se
+            $query = "SELECT tilausrivilinkki
+                      FROM tilausrivin_lisatiedot
+                      WHERE yhtio = '{$kukarow['yhtio']}'
+                      AND tilausrivitunnus = '{$rivitunnus}'";
+            $result = pupe_query($query);
+            $_myyntirivi = mysql_fetch_assoc($result);
+
+            if (mysql_num_rows($result) == 1 and $_myyntirivi["tilausrivilinkki"] != 0) {
+              // Tarkistetaan, ett� ostotilaus on viel� kesken,
+              // koska jos ei ole kesken on jo l�hetetty eteenp�in
+              $query = "SELECT lasku.tila,
+                        lasku.alatila,
+                        lasku.tunnus
+                        FROM tilausrivi
+                          JOIN lasku ON (lasku.yhtio = tilausrivi.yhtio AND lasku.tunnus = tilausrivi.otunnus)
+                        WHERE tilausrivi.yhtio = '{$kukarow['yhtio']}'
+                        AND tilausrivi.tunnus = '{$_myyntirivi["tilausrivilinkki"]}'";
```

Comment:
@hannahagberg Tilausrivillä pitää olla ehtoina myös ne, että rivi on ostorivi ja sitä ei ole vielä viety varastoon.

Itse oisin kirjoittanut tän myös suoraan deletenä. Vähemmän koodia ja mielestäni selkeämpi:

DELETE tilausrivi.*
JOIN lasku on (jhs jshd jsdh jdh )
WHERE tilausrivi.kshdkjsjh = 'kala'
and
and 
and